src/Trinity.C/src/Storage/MemoryTrunk/MemoryTrunk.Memory.cpp

```diff
@@ -250,8 +263,7 @@ namespace Storage
         size_t append_head = cp - buffer;
         if (Memory::MemoryCommit(trunkPtr, append_head) == NULL)
         {
-            free(buffer);
-            return -1;
+            Trinity::Diagnostics::FatalError("ReloadImpl: MemoryTrunk {0} fails to reclaim memory after data's been moved to a temporary buffer.", TrunkId);
```

Comment:
Why we dont free(buffer) and return now?

---
In reply to: [105527550](https://github.com/Microsoft/GraphEngine/pull/53#discussion_r105527550) [](ancestors = 105527550)